### ISSUE TITLE
Allow passing typed search attributes to child workflow

### ISSUE THREAD

**Issue by oridag** [vitaliihonta/zio-temporal#116]
With the Java SDK one can do:
```scala
  val sa = Workflow.getTypedSearchAttributes
  val optionsBuilder = ChildWorkflowOptions.newBuilder().setTypedSearchAttributes(sa)
```

With the current zio-temporal API, this is not possible, or at least I couldn't find a way to do that

**Comment by vitaliihonta** [comment]
TODO: Add `ZChildWorkflowStubBuilder.withSearchAttributes` method overload that accepts plain `SearchAttributes` (not a Map later encoded by zio-temporal).
Other methods may require such an overload as well

**Comment by grouzen** [comment]
I think I can work on it.

**Comment by vitaliihonta** [comment]
@grouzen Cool, I appreciate your help! Here are a few links to simplify your work:
1. I'd use `ZSearchAttributes` [wrapper](https://github.com/vitaliihonta/zio-temporal/blob/main/core/src/main/scala/zio/temporal/ZSearchAttributes.scala) instead of the java class.
2. To make the interop better, we might add `ZSearchAttributes.fromJava` method that allows passing search attributes constructed via Java SDK directly (in case teams have mixed Java/Scala code).
3. It would be great to add this overload to other places as well (for instance, [here](https://github.com/vitaliihonta/zio-temporal/blob/main/core/src/main/scala/zio/temporal/workflow/ZWorkflowStubBuilder.scala#L67))

**Comment by grouzen** [comment]
Great! Thanks for the clear instructions!

**Comment by vitaliihonta** [comment]
@oridag, how urgently do you need this to be released? The next planned `0.5.0` release is pretty small and contains a single little breaking change related to protobuf payload converter (as we dropped an old java dependency).

My upcoming work on #121 breaks a lot to support more Java SDK functionality. Therefore, I wonder what's the right time to release this feature

**Comment by oridag** [comment]
I don't mind waiting a week or two, but if it's longer than that I'd appreciate a release, thanks!
